source/megengine.js

```diff
@@ -465,6 +656,226 @@ megengine.Tensor = class {
         return this._type;
     }
 
+    get format() {
+        return this._format;
+    }
+
+    get state() {
+        return this._context().state;
+    }
+
+    get value() {
+        const context = this._context();
+        if (context.state) {
+            return null;
+        }
+        context.limit = Number.MAX_SAFE_INTEGER;
+        return this._decode(context, 0);
+    }
+
+    toString() {
+        const context = this._context();
+        if (context.state) {
+            return '';
+        }
+        context.limit = 10000;
+        const value = this._decode(context, 0);
+        return megengine.Tensor._stringify(value, '', '    ');
+    }
+
+    _context() {
+        const context = {};
+        context.state = null;
+        context.index = 0;
+        context.count = 0;
+
+        if (!this._type.dataType) {
+            context.state = 'Tensor has no data type.';
+            return context;
+        }
+        switch (this._type.dataType) {
+            case 'bool':
```

Comment:
Remove and update. Other formats now use shared rendering code in `sidebar.Tensor`. `dataType` names should map to the general names. For example, `bool` is not a valid name, should be `boolean`.

Reply:
done